plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/validators/N4JSExpressionValidator.xtend

```diff
@@ -874,38 +876,51 @@ class N4JSExpressionValidator extends AbstractN4JSDeclarativeValidator {
 	 */
 	@Check
 	def checkAdditiveExpressionForNonADDs(AdditiveExpression ae) {
-
-		if (ae.rhs === null || ae.lhs === null) {
+		if (ae.op == AdditiveOperator.SUB) {
+			doCheckMathOperands(ae, ae.rhs, ae.lhs, "substraction");
+		}
+	}
+	
+	def doCheckMathOperands(Expression expr, Expression lhs, Expression rhs, String mathExpr) {
+		if (rhs === null || lhs === null) {
 			return; // corrupt AST (e.g., while editing)
 		}
+		
+		// The types of the operands must be subtypes of number if the operator is not ’+’
+		val bits = BuiltInTypeScope.get(expr.eResource.resourceSet)
 
-		if (ae.op !== AdditiveOperator.ADD) {
-
-			// The types of the operands must be subtypes of number if the operator is not ’+’
-			val bits = BuiltInTypeScope.get(ae.eResource.resourceSet)
-
-			val tlhs = ts.tau(ae.lhs)
-			if (tlhs === null) {
-				return; // corrupt AST (e.g., while editing)
-			}
-			if (tlhs.declaredType === bits.nullType || tlhs.declaredType === bits.undefinedType)
-				issueNotANumberType(tlhs.declaredType.name, ae.lhs);
-
-			val trhs = ts.tau(ae.rhs)
-			if (trhs === null) {
-				return; // corrupt AST (e.g., while editing)
-			}
-
-			if (trhs.declaredType === bits.nullType || trhs.declaredType === bits.undefinedType)
-				issueNotANumberType(trhs.declaredType.name, ae.rhs);
+		val tlhs = ts.tau(lhs)
+		if (tlhs === null) {
+			return; // corrupt AST (e.g., while editing)
+		}
+		if (tlhs.declaredType === bits.undefinedType)
```

Comment:
why not already support value `null` as well (i.e. show warning if declaredType if `null` type)?

Reply:
I added a waring that null is interpreted as 0. Note that

```
1/null != 0/null
```